src/main/java/de/upb/crypto/math/structures/groups/elliptic/nopairing/Secp256k1.java

```diff
@@ -159,4 +168,46 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return 0;
     }
+
+    public static class HashIntoSecp256k1 implements HashIntoGroupImpl {
```

Comment:
why inner class?

Reply:
this means it can only be instantiated if you have a instance of Secp256k1. Is this intended?

Reply:
it's a static inner class, so it can be instantiated without an instance of `Secp256k1`. 

Inner class was just in order to keep the number of files down. No strong feelings about this.

Reply:
Is this something I should change?

Reply:
Your constructor thinks that zp is always instantiated, but without an instance of Secp256k1, why should it be?

Reply:
ah zp is static.